src/test/scala/org/broadinstitute/dsde/workbench/sam/azure/AzureServiceSpec.scala

```diff
@@ -107,13 +139,219 @@ class AzureServiceSpec(_system: ActorSystem) extends TestKit(_system) with AnyFl
+    val resourceName = ResourceId("resource")
+    val resource = FullyQualifiedResourceId(defaultResourceType.name, resourceName)
+    resourceService.createResourceType(defaultResourceType, samRequestContext).unsafeRunSync()
+    runAndWait(resourceService.createResource(defaultResourceType, resourceName, defaultUser, samRequestContext))
+
+    // Create the "billing profile" resource. There's no actual need for it to be a "billing profile", we just need a resource to attach the managed resource group to.
+    val billingProfileId = BillingProfileId("de38969d-f41b-4b80-99ba-db481e6db1cf")
+    val billingProfileResource = runAndWait(resourceService.createResource(defaultResourceType, billingProfileId.asResourceId, defaultUser, samRequestContext))
+
+    val tenantId = TenantId(azureTestConfig.getString("tenantId"))
+    val subscriptionId = SubscriptionId(azureTestConfig.getString("subscriptionId"))
+    val managedResourceGroupName = ManagedResourceGroupName(azureTestConfig.getString("managedResourceGroupName"))
+    val mrgCoordinates = ManagedResourceGroupCoordinates(tenantId, subscriptionId, managedResourceGroupName)
+    val managedResourceGroup = ManagedResourceGroup(mrgCoordinates, billingProfileId)
+    runAndWait(azureService.createManagedResourceGroup(managedResourceGroup, samRequestContext.copy(samUser = Some(defaultUser))))
+
+    val actionManagedIdentityId =
+      ActionManagedIdentityId(resource, viewAction, billingProfileId)
+
+    // create action managed identity
+    val (res, created) = azureService
+      .getOrCreateActionManagedIdentity(resource, viewAction, billingProfileId, samRequestContext.copy(samUser = Some(defaultUser)))
+      .unsafeRunSync()
+    created shouldBe true
+    res.id shouldBe actionManagedIdentityId
+    res.displayName shouldBe ManagedIdentityDisplayName(s"${resource.resourceId.value}-${viewAction.value}")
+
+    // managed identity should now exist in azure
+    val msiManager = crlService.buildMsiManager(tenantId, subscriptionId).unsafeRunSync()
+    val azureRes = msiManager.identities().getById(res.objectId.value)
+    azureRes should not be null
+    azureRes.tenantId() shouldBe tenantId.value
+    azureRes.resourceGroupName() shouldBe managedResourceGroupName.value
+    azureRes.id() shouldBe res.objectId.value
+    azureRes.name() shouldBe res.displayName.value
+
+    // delete action managed identity
+    azureService.deleteActionManagedIdentity(actionManagedIdentityId, samRequestContext).unsafeRunSync()
+
+    // action managed identity should not exist in postgres
+    directoryDAO.loadActionManagedIdentity(actionManagedIdentityId, samRequestContext).unsafeRunSync() shouldBe None
+
+    // managed identity should not exist in Azure
+    msiManager.identities().listByResourceGroup(managedResourceGroupName.value).asScala.toList.exists { i =>
+      i.name() === azureService.toManagedIdentityNameFromAmiId(actionManagedIdentityId)
+    } shouldBe false
+  }
```

Comment:
This delete test could maybe just be merged with the above creation test? feels like there is a lot of overlap. Up to you though.